packages/web/components/Header/NewTechnologyButton.js

```diff
@@ -0,0 +1,36 @@
+import React from 'react';
+import { useRouter } from 'next/router';
+import { useTranslation } from 'react-i18next';
+import { Button } from './styles';
+import { useModal, useAuth } from '../../hooks';
+
+const NewTechnologyButton = () => {
+	const { t } = useTranslation(['common']);
+	const { openModal } = useModal();
+	const { user } = useAuth();
+	const router = useRouter();
+	const url = '/technology/new';
+	const handleClick = (e) => {
+		e.preventDefault();
+		if (!user.email) {
+			openModal('login', {
+				message: t('common:signInToContinue'),
+				redirectTo: url,
+			});
+		} else if (router.asPath !== url) {
+			router.push(url);
+		}
+	};
+	return (
+		<Button onClick={handleClick}>
```

Comment:
Use o component `Link` do `next/link` e use o onClick handler no Link.

Tecnicamente o correto é usar o Link do `next/link`, pois assim o link funcionaria mesmo sem JavaScript.

Reply:
Além disso, em `pages/technology/new.js` faça uma verificação naquela página para não renderizar nada caso o usuário não esteja logado.

Reply:
O component `Link` do `next/link` não tem a `prop` `onClick`, esse `Button` é uma tag 'a' estilizada. A ideia é manipular o valor do href ?

Reply:
o `onClick` é no filho dele: https://nextjs.org/docs/api-reference/next/link#using-a-component-that-supports-onclick. Nesse caso, seria na tag `a` estilizada.